modules/composer-ffmpeg/src/main/java/org/opencastproject/composer/impl/EncoderEngine.java

```diff
@@ -392,4 +461,526 @@ private void handleEncoderOutput(List<File> output, String message) {
+    private final ArrayList<String> afilter;
+    private String vInputPad = "";
+    private String aInputPad = "";
+    private String vsplit = "";
+    private String asplit = "";
+
+    /*
+     * set the audio filter if there are any in the profiles or identity
+     */
+    private void setAudioFilters() {
+      if (pf.size() == 1) {
+        if (afilter.get(0) != null)
+          afilter.set(0, aInputPad + afilter.get(0) + apads.get(0)); // Use audio filter on input directly
+      } else
+        for (int i = 0; i < pf.size(); i++) {
+          if (afilter.get(i) != null) {
+            afilter.set(i, "[oa0" + i + "]" + afilter.get(i) + apads.get(i)); // Use audio filter on apad
+            asplit += "[oa0" + i + "]";
+          } else {
+            asplit += apads.get(i); // straight to output
+          }
+        }
+      afilter.removeAll(Arrays.asList((String) null));
+    }
+
+    /*
+     * set the video filter if there are any in the profiles
+     */
+    private void setVideoFilters() {
+      if (pf.size() == 1) {
+        if (vfilter.get(0) != null)
+          vfilter.set(0, vInputPad + vfilter.get(0) + vpads.get(0)); // send to filter first
+      } else
+        for (int i = 0; i < pf.size(); i++) {
+          if (vfilter.get(i) != null) {
+            vfilter.set(i, "[ov0" + i + "]" + vfilter.get(i) + vpads.get(i)); // send to filter first
+            vsplit += "[ov0" + i + "]";
+          } else {
+            vsplit += vpads.get(i);// straight to output
+          }
+        }
+
+      vfilter.removeAll(Arrays.asList((String) null));
+    }
+
+    public List<String> getOutFiles() {
+      return outputFiles;
+    }
+
+    /**
+     *
+     * @return output pads - the "-map xyz" clauses
+     */
+    public List<String> getOutput() {
+      return outputs;
+    }
+
+    /**
+     *
+     * @return filter split clause for ffmpeg
+     */
+    public String getVsplit() {
+      return vsplit;
+    }
+
+    public String getAsplit() {
+      return asplit;
+    }
+
+    public String getVideoFilter() {
+      if (vfilter.isEmpty())
+        return null;
+      return StringUtils.join(vfilter, ";");
+    }
+
+    public String getAudioFilter() {
+      if (afilter.isEmpty())
+        return null;
+      return StringUtils.join(afilter, ";");
+    }
+
+    /**
+     * If this is a raw mapping not used with complex filter, strip the square brackets if there are any
+     *
+     * @param pad
+     *          - such as 0:a, [0:v], [1:1],[0:12],[main],[overlay]
+     * @return adjusted pad
+     */
+    public String adjustForNoComplexFilter(String pad) {
+      final Pattern outpad = Pattern.compile("\\[(\\d+:[av\\d{1,2}])\\]");
+      try {
+        Matcher matcher = outpad.matcher(pad); // throws exception if pad is null
+        if (matcher != null && matcher.matches()) {
+          logger.info(pad + matcher.toString());
+          return matcher.group(1);
+        }
+      } catch (Exception e) {
```

Comment:
The code in the `try` block should not throw any exceptions, should it?

Reply:
It does when pad is null.